idea/idea-fir/src/org/jetbrains/kotlin/idea/quickfix/fixes/ReplaceCallFixFactories.kt

```diff
@@ -34,16 +31,47 @@ object ReplaceCallFixFactories {
                 psi
             }.unwrapParenthesesLabelsAndAnnotations()
 
+            val shouldHaveNotNullType = target.safeAs<KtExpression>()?.shouldHaveNotNullType(this) ?: false
             when (target) {
-                is KtDotQualifiedExpression -> listOf(ReplaceWithSafeCallFix(target, target.shouldHaveNotNullType()))
+                is KtDotQualifiedExpression -> listOf(ReplaceWithSafeCallFix(target, shouldHaveNotNullType))
                 is KtNameReferenceExpression -> {
                     // TODO: As a safety precaution, resolve the expression to determine if it is a call with an implicit receiver.
                     // This is a defensive check to ensure that the diagnostic was reported on such a call and not some other name reference.
                     // This isn't strictly needed because FIR checkers aren't reporting on wrong elements, but ReplaceWithSafeCallFixFactory
                     // in FE1.0 does so.
-                    listOf(ReplaceImplicitReceiverCallFix(target, target.shouldHaveNotNullType()))
+                    listOf(ReplaceImplicitReceiverCallFix(target, shouldHaveNotNullType))
                 }
+                is KtArrayAccessExpression -> listOf(ReplaceInfixOrOperatorCallFix(target, shouldHaveNotNullType))
                 else -> emptyList()
             }
         }
+
+    val unsafeInfixCallFactory =
+        diagnosticFixFactory<KtFirDiagnostic.UnsafeInfixCall> { diagnostic ->
+            val psi = diagnostic.psi
+            val target = psi.parent as? KtBinaryExpression ?: return@diagnosticFixFactory emptyList()
+            listOf(ReplaceInfixOrOperatorCallFix(target, target.shouldHaveNotNullType(this), diagnostic.operator))
+        }
+
+    val unsafeOperatorCallFactory =
+        diagnosticFixFactory<KtFirDiagnostic.UnsafeOperatorCall> { diagnostic ->
+            val psi = diagnostic.psi
+            val operationToken = psi.safeAs<KtOperationReferenceExpression>()?.getReferencedNameElementType()
+            if (operationToken == KtTokens.EQ || operationToken in OperatorConventions.COMPARISON_OPERATIONS) {
+                // This matches FE1.0 behavior; see ReplaceInfixOrOperatorCallFixFactory.kt
+                return@diagnosticFixFactory emptyList()
+            }
+
+            val target = psi.parent as? KtBinaryExpression ?: return@diagnosticFixFactory emptyList()
+            listOf(ReplaceInfixOrOperatorCallFix(target, target.shouldHaveNotNullType(this), diagnostic.operator))
+        }
+
+    private fun KtExpression.shouldHaveNotNullType(session: KtAnalysisSession): Boolean {
```

Comment:
`KtAnalysisSession` should be always passed as receiver parameter as a check that it is not leaked outside analysis scope

Reply:
Thanks for the reminder. Fixed